quantization/CentroidsGenerator.h

```diff
@@ -92,20 +96,15 @@ class CentroidsGenerator {
    */
   void generateCentroids(
       const float *vectors, const float *vec_weights, uint64_t n,
-      std::function<float(const float *, const float *)> &distance_func) {
+      const std::function<float(const float *, const float *)> &distance_func) {
     if (n < _num_centroids) {
       throw std::runtime_error(
           "Invalid configuration. The number of centroids: " +
           std::to_string(_num_centroids) +
           " is bigger than the number of data points: " + std::to_string(n));
     }
 
-    // Initialize the centroids by randomly sampling k centroids among the n
-    // data points
-    if (!_centroids_initialized) {
```

Comment:
This is essentially what was causing poor recall (although there might still be another bug source since the recall is not where we want just yet). This check meant that we were running clustering only on the first sub-quantizer.